packages/jest-jasmine2/src/jestExpect.ts

```diff
@@ -41,23 +41,15 @@ export default function jestExpect(config: {expand: boolean}): void {
       jestMatchersObject[name] = function (
         this: MatcherState,
         ...args: Array<unknown>
-      ): RawMatcherFn {
```

Comment:
This `function` has inferred return type of `ExpectationResult` without this extra type.